packages/protocol/scripts/foundry/create_and_migrate_anvil_l2_devchain.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Read environment variables and constants
+source $PWD/scripts/foundry/constants.sh
+
+# Generate and run L1 devchain
+echo "Generating and running L1 devchain before activating L2..."
+source $PWD/scripts/foundry/create_and_migrate_anvil_devchain.sh
+
+# Stop L1 devchain
+source $PWD/scripts/foundry/stop_anvil.sh
+
+# Duplicate L1 state
+cp $TMP_FOLDER/$L1_DEVCHAIN_FILE_NAME $TMP_FOLDER/$L2_DEVCHAIN_FILE_NAME
+
+# Start L2 devchain from L1 state and apply L2 migrations
+echo "Anvil L2 state will be saved to $TMP_FOLDER/$L2_DEVCHAIN_FILE_NAME"
+anvil \
+--port $ANVIL_PORT \
+--state $TMP_FOLDER/$L2_DEVCHAIN_FILE_NAME \
+--state-interval $STATE_INTERVAL \
+--gas-limit $GAS_LIMIT \
+--code-size-limit $CODE_SIZE_LIMIT \
+--balance $BALANCE \
+--steps-tracing &
+
+# Activate the L2 on the devchain by deploying arbitrary bytecode to the PROXY_ADMIN_ADDRESS
+# Arbitrarily using Registry.sol bytcode, but we could have used any 
+# other arbitary bytecode of length > 0. See `isL2()` implementation in IsL2Check.sol.
+echo "Activating L2 on the devchain..."
+REGISTRY_BYTECODE=$(jq -r '.bytecode.object' ./out/Registry.sol/Registry.json)
+cast rpc anvil_setCode \
+$PROXY_ADMIN_ADDRESS \
+$REGISTRY_BYTECODE \
+--rpc-url $ANVIL_RPC_URL
+
+# Fetch address of Celo distribution 
+CELO_DISTRIBUTION_SCHEDULE_ADDRESS=$(
```

Comment:
I'd replace this file with a forge script (like Migrations.s.sol), so we have all this code already written in solidity and we can combine it to just one migration after the mainnet transition is done.

Reply:
Great point, I propose:
1. we merge this PR as is, that way DevTooling can get started using the L2 devchain. 
2. we open an issue to refactor this workflow into a Foundry script.

I'll create an issue to do this.